tests/unit/conftest.py

```diff
@@ -83,6 +84,14 @@ def lti_v13_params():
     }
 
 
+@pytest.fixture
+def with_plugins(pyramid_request, mock_service):
+    # Ensure that when plugins are read from the current product, all of its
+    # plugins are mocked and ready to be looked up
+    for plugin_class in asdict(pyramid_request.product.plugin_config).values():
+        mock_service(plugin_class)
```

Comment:
You can't automatically do this as a part of the `pyramid_request` as `mock_service` requires `pyramid_config` which requires `pyramid_request`...

Due to our lazy loading mechanism, we don't need to do this up front. So we can create the product object, then register the mocks before they are called.